src/Apps/W1/PEPPOL/App/src/Common/PEPPOL30.Codeunit.al

```diff
@@ -0,0 +1,1115 @@
+// ------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+// ------------------------------------------------------------------------------------------------
+namespace Microsoft.Peppol;
+
+using Microsoft.Finance.VAT.Calculation;
+using Microsoft.Finance.VAT.Setup;
+using Microsoft.Foundation.Attachment;
+using Microsoft.Sales.Document;
+using Microsoft.Sales.History;
+using Microsoft.Service.Document;
+
+codeunit 37200 "PEPPOL30" implements "PEPPOL Attachment Handler"
+                                            , "PEPPOL Delivery Info Provider"
+                                            , "PEPPOL Document Info Provider"
+                                            , "PEPPOL Line Info Provider"
+                                            , "PEPPOL Monetary Info Provider"
+                                            , "PEPPOL Party Info Provider"
+                                            , "PEPPOL Payment Info Provider"
+                                            , "PEPPOL Tax Info Provider"
+{
+
+    InherentEntitlements = X;
+    InherentPermissions = X;
+
+    var
+        PEPPOLManagementImpl: Codeunit "PEPPOL30 Impl.";
+
+    /// <summary>
+    /// Gets general invoice information including ID, issue date, invoice type, currency codes, and accounting cost.
+    /// </summary>
+    /// <param name="SalesHeader">The sales header record containing the document information.</param>
+    /// <param name="FormatProvider">The format provider interface for PEPPOL formatting.</param>
+    /// <param name="ID">Returns the invoice ID.</param>
+    /// <param name="IssueDate">Returns the invoice issue date.</param>
+    /// <param name="InvoiceTypeCode">Returns the invoice type code.</param>
+    /// <param name="InvoiceTypeCodeListID">Returns the invoice type code list ID.</param>
+    /// <param name="Note">Returns any additional notes.</param>
+    /// <param name="TaxPointDate">Returns the tax point date.</param>
+    /// <param name="DocumentCurrencyCode">Returns the document currency code.</param>
+    /// <param name="DocumentCurrencyCodeListID">Returns the document currency code list ID.</param>
+    /// <param name="TaxCurrencyCode">Returns the tax currency code.</param>
+    /// <param name="TaxCurrencyCodeListID">Returns the tax currency code list ID.</param>
+    /// <param name="AccountingCost">Returns the accounting cost reference.</param>
+    procedure GetGeneralInfo(SalesHeader: Record "Sales Header"; var ID: Text; var IssueDate: Text; var InvoiceTypeCode: Text; var InvoiceTypeCodeListID: Text; var Note: Text; var TaxPointDate: Text; var DocumentCurrencyCode: Text; var DocumentCurrencyCodeListID: Text; var TaxCurrencyCode: Text; var TaxCurrencyCodeListID: Text; var AccountingCost: Text)
```

Comment:
There are many parameters, maybe you could get the argument table here or a codeunit that is representing this object?